lib/src/obsoleted/analyzer_plugin/analyzer_plugin.dart

```diff
@@ -100,20 +101,27 @@ class MetricsAnalyzerPlugin extends ServerPlugin {
       return dartDriver;
     }
 
+    final analysisOptionsFilePath = contextRoot.optionsFile;
+    // We need to take into account mono repositories / multiroot workspaces where
+    // "rootPath" is not the same as "analysisOptions.yaml" file location.
+    final excludeRootPath = analysisOptionsFilePath == null
```

Comment:
`contextRoot,optionsFile` is nullable field but I do not know scenarios when it can be null. In this case `rootPath` is still used as root for excludes but I am not sure if this is correct.